task.py

```diff
@@ -29,47 +31,86 @@ def main():
                 raise
             os.remove('/tmp/previewer/' + file)
 
+def cleanup_past_run(directory):
+    if os.path.isdir(directory) == True:
+        project = get_project(directory)
+        project.kill()
+        project.remove_stopped()
+        shutil.rmtree(directory)
+
+def run_docker_compose(networkPrefix, environment, workingDirectory):
+    dockerHelper = DockerHelper()
+    dockerHelper.clean_old_images()
+    dockerHelper.pull_container_image('jwilder/nginx-proxy')
+    dockerHelper.create_network(networkPrefix + '_default')
+    containerArgs = ["-v", "/var/run/docker.sock:/tmp/docker.sock:ro", "-p", "8111:80"]
+    dockerHelper.run_container('nginx-proxy', 'jwilder/nginx-proxy', containerArgs)
+    dockerHelper.container_join_network(networkPrefix + '_default', 'nginx-proxy')
+    
+    os.environ = environment
+    project = get_project(workingDirectory)
+    project.pull()
+    project.build()
+    project.up(detached=True)
+    
+    dockerHelper.prune_all()
+
+
+def branch(data):
+    branchName = str(data['ref']).split("/", 2)[-1]
+    safeBranchName = safeRegexPattern.sub('', str(data['ref']).split("/")[-1])
+    workingDirectory = '/tmp/' + safeBranchName
+    subDomain = '.' + data['repository']['name'] + '.previewer.mashape.com'
+    
+    cleanup_past_run(workingDirectory)
```

Comment:
What happens when an error occurs?

Reply:
All errors are caught and we delete the task file that defines the work.

So the task that caused the error will be in an unknown state, the task runner will continue to run and will continue on subsequent tasks. 

It's a unwieldy way to handle errors but my thoughts were do everything possible to keep it processing the tasks and then when an exception needs to be made and we have to handle an error more elegantly do it then.

Reply:
Alright, sounds good to me, looks good to me, would love for the github message to be a constant that is a formatted string instead:

`The preview is available at: %s`

for example, other than that, looks good